arwen/host/execution.go

```diff
@@ -117,29 +119,39 @@ func (host *vmHost) doRunSmartContractCall(input *vmcommon.ContractCallInput) (v
 	}()
 
 	result, returnCode, err := host.callSCMethod()
+
+	metering.UnlockGasIfAsyncStep()
+
 	if err != nil {
 		return vmOutput
 	}
 
-	vmOutput = output.GetVMOutput(result)
+	output.FinishValue(result)
+	vmOutput = output.GetVMOutput()
 
 	return vmOutput
 }
 
-func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) error {
+func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
 	host.PushState()
 
 	host.InitState()
 
 	host.Runtime().InitStateFromContractCallInput(input)
 	err := host.execute(input)
 
+	var vmOutput *vmcommon.VMOutput
+	vmOutput = nil
```

Comment:
declare vmOutput before error. No need to make vmOutput nil. return nil, err if err != nil.

Reply:
Changed.